extensions/MongoDbAtlas/MongoDbAtlas/MongoDbAtlasMemory.cs

```diff
@@ -0,0 +1,351 @@
+        await this._utils.WaitForIndexToBeReadyAsync(collectionName, 120).ConfigureAwait(false);
+
+        // Keep tracks of created indexes.
+        var collection = this.Database.GetCollection<BsonDocument>(GetIndexListCollectionName());
+        // upsert the name of the index
+        var filter = Builders<BsonDocument>.Filter.Eq("_id", normalizedIndexName);
+        var update = Builders<BsonDocument>.Update.Set("index", normalizedIndexName).Set("lastCreateIndex", DateTime.UtcNow);
+        await collection.UpdateOneAsync(filter, update, new UpdateOptions { IsUpsert = true }, cancellationToken).ConfigureAwait(false);
+    }
+
+    public async Task DeleteIndexAsync(string index, CancellationToken cancellationToken = default)
+    {
+        var normalizedIndexName = NormalizeIndexName(index);
+        if (this.Config.UseSingleCollectionForVectorSearch)
+        {
+            // Actually if we use a single collection we do not delete the entire collection we simply delete records of the index
+            var collection = this.GetCollectionFromIndexName(index);
+            await collection.DeleteManyAsync(x => x.Index == normalizedIndexName, cancellationToken: cancellationToken).ConfigureAwait(false);
+        }
+        else
+        {
+            var collectionName = this.GetCollectionName(index);
+            await this._utils.DeleteIndicesAsync(collectionName).ConfigureAwait(false);
+            await this.Database.DropCollectionAsync(collectionName, cancellationToken).ConfigureAwait(false);
+        }
+
+        var collectionIndexList = this.Database.GetCollection<BsonDocument>(GetIndexListCollectionName());
+        await collectionIndexList.DeleteOneAsync(x => x["_id"] == index, cancellationToken: cancellationToken).ConfigureAwait(false);
+    }
+
+    public async Task<IEnumerable<string>> GetIndexesAsync(CancellationToken cancellationToken = default)
+    {
+        // We load index from the index list collection
+        var collection = this.Database.GetCollection<BsonDocument>(GetIndexListCollectionName());
+        var cursor = await collection.FindAsync(Builders<BsonDocument>.Filter.Empty, cancellationToken: cancellationToken).ConfigureAwait(false);
+        return cursor.ToEnumerable(cancellationToken: cancellationToken).Select(x => x["_id"].AsString).ToImmutableArray();
+    }
+
+    public Task DeleteAsync(string index, MemoryRecord record, CancellationToken cancellationToken = default)
+    {
+        var collection = this.GetCollectionFromIndexName(index);
+        return collection.DeleteOneAsync(x => x.Id == record.Id, cancellationToken: cancellationToken);
+    }
+
+    public async IAsyncEnumerable<MemoryRecord> GetListAsync(
+        string index,
+        ICollection<MemoryFilter>? filters = null,
+        int limit = 1,
+        bool withEmbeddings = false,
+        [EnumeratorCancellation] CancellationToken cancellationToken = default)
+    {
+        if (limit <= 0)
+        {
+            limit = 10;
```

Comment:
this var is never used, the method should stop after returning `limit` records

Reply:
I've put a check inside the loop, but this value is passed to the method that created the filter, the number of item is directly inserted in the mongodb query.